packages/block/src/header.ts

```diff
@@ -233,13 +243,13 @@ export class BlockHeader {
       if (timestamp.isZero()) {
         timestamp = new BN(toBuffer(this._common.genesis().timestamp))
       }
-      if (difficulty.isZero()) {
+      if (difficulty.isZero() && !this._common.isActivatedEIP(3675)) {
```

Comment:
i think if you were to put `// Set these values as constants after the Merge` after this `if (options.initWithGenesisHeader)` block then you wouldn't need to add these additional  `&& !this._common.isActivatedEIP(3675)` since the fields will simply be overwritten.

Reply:
Update: This also goes for the adopted logic of throwing on wrong values. Apart from the question if/when there will be such thing like a PoS genesis header: if such a header *is defined including* the respective constant values, these will for sure match the anticipated values. If these values are left out, they will be filled up by our library with the correct values.

Will remove the checks.
---

Side note: I actually got a bit unsure again along the way when doing b9e9625 but this is confirmed by the spec:

> Add verification of the fields noted in block structure as their specified constant value, including the enforcement of empty ommers list.